tfhe/src/integer/server_key/radix/tests/modulus_switch_compression.rs

```diff
@@ -0,0 +1,61 @@
+use crate::integer::gen_keys_radix;
+use crate::integer::tests::create_parametrized_test;
+#[cfg(tarpaulin)]
+use crate::shortint::parameters::coverage_parameters::*;
+use crate::shortint::parameters::*;
+use rand::Rng;
+
+const NB_TESTS: usize = 10;
+
+create_parametrized_test!(modulus_switch_compression);
+create_parametrized_test!(modulus_switch_compression_signed);
+
+fn modulus_switch_compression<P>(param: P)
+where
+    P: Into<PBSParameters>,
+{
+    // We have 4 * 2 = 8 bits of message
+    let size = 4;
+    let (cks, sks) = gen_keys_radix(param.into(), size);
+
+    let mut rng = rand::thread_rng();
+
+    for _ in 0..NB_TESTS {
+        let clear = rng.gen::<u8>();
+
+        let ctxt = cks.encrypt(clear);
+
+        let compressed_ct = sks.switch_modulus_and_compress(&ctxt);
+
+        let decompressed_ct = sks.decompress(&compressed_ct);
+
+        let dec: u8 = cks.decrypt(&decompressed_ct);
+
+        assert_eq!(clear, dec);
+    }
+}
+
+fn modulus_switch_compression_signed<P>(param: P)
+where
+    P: Into<PBSParameters>,
```

Comment:
don't know if that's the pattern we use for test organization in integer, could you check with Thomas please ?